econplayground/main/views.py

```diff
@@ -60,6 +64,79 @@ def test_func(self):
         return user_is_instructor(self.request.user)
 
 
+class GraphCloneDisplay(LoginRequiredMixin, CohortInstructorMixin, DetailView):
+    model = Graph
+    template_name = 'main/graph_clone_form.html'
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+            'form': GraphCloneForm(user=self.request.user),
+        })
+        return ctx
+
+
+class GraphCloneFormView(LoginRequiredMixin, CohortInstructorMixin,
+                         SingleObjectMixin, FormView):
+    template_name = 'main/graph_clone_form.html'
+    form_class = GraphCloneForm
+    model = Graph
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.request.user
+        return kwargs
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+            'form': GraphCloneForm(user=self.request.user),
+        })
+        return ctx
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().post(request, *args, **kwargs)
+
+    def get_success_url(self):
+        return reverse('cohort_detail', kwargs={'pk': self.cohort.pk})
+
+    def form_valid(self, form):
+        cohort_pk = int(form.data.get('course'))
+        cohort = Cohort.objects.get(pk=cohort_pk)
+        g_topic = cohort.get_general_topic()
+
+        cloned = self.object.clone()
+        cloned.title = '{} (clone)'.format(self.object.title)
+        cloned.author = self.request.user
+        cloned.topic = g_topic
+        cloned.is_published = False
```

Comment:
Noting that the "featured" flag is also carried over on clone. That may be desired behavior, but wanted to point it out. (I'm going ahead with the merge now.)